lib/enrichment/src/tables.rs

```diff
@@ -178,6 +178,26 @@ impl TableSearch {
             Err("finish_load not called".to_string())
         }
     }
+
+    /// Search the enrichment table data with the given condition.
+    /// All conditions must match (AND).
+    /// Can return multiple matched records
+    pub fn find_table_rows<'a>(
```

Comment:
Given this new function, I'm wondering if `find_table_row` shouldn't just call it and error if there are more than one.

Reply:
There is a subtle difference in that `find_table_row` will  return with an error as soon as it loads a second row. This saves allocating any more than it needs to. 

It is a trade off with code complexity especially when the reduced performance comes from an error condition, so if there were strong feeling I'd consider changing it. But for now,  my vote is towards performance.